source/Calamari.Testing/EnvironmentVariables.cs

```diff
@@ -70,31 +77,46 @@ public enum ExternalVariable
         [EnvironmentVariable("CALAMARI_AUTHPASSWORD", "OctopusMyGetTester")]
         MyGetFeedPassword,
 
-        [EnvironmentVariable("GOOGLECLOUD_OCTOPUSAPITESTER_JSONKEY", "GoogleCloud - OctopusAPITester")]
+        [EnvironmentVariable("GOOGLECLOUD_OCTOPUSAPITESTER_JSONKEY", "GoogleCloud - OctopusAPITester", "op://Calamari Secrets for Tests/Google Cloud Octopus Api Tester JsonKey")]
         GoogleCloudJsonKeyfile,
-        
+
         [EnvironmentVariable("GitHub_RateLimitingPersonalAccessToken", "GitHub test account PAT")]
         GitHubRateLimitingPersonalAccessToken,
     }
 
     public static class ExternalVariables
     {
+        static readonly bool SecretManagerIsEnabled = Convert.ToBoolean(Environment.GetEnvironmentVariable("CALAMARI__Tests__SecretManagerEnabled") ?? "True");
+        static readonly string SecretManagerAccount = Environment.GetEnvironmentVariable("CALAMARI__Tests__SecretManagerAccount") ?? "octopusdeploy.1password.com";
+
+        static readonly Lazy<SecretManagerClient> SecretManagerClient = new(LoadSecretManagerClient);
+
+        static SecretManagerClient LoadSecretManagerClient()
+        {
+            var allVariableAttributes = EnvironmentVariableAttribute.GetAll()
+                                                                    .Select(x => x.SecretReference)
+                                                                    .Where(x => x is not null)
+                                                                    .ToArray();
+            var loggerFactory = new LoggerFactory(); //.AddSerilog(Logger);
+            var microsoftLogger = loggerFactory.CreateLogger<SecretManagerClient>();
+            return new SecretManagerClient(SecretManagerAccount, allVariableAttributes, microsoftLogger);
+        }
+
         public static void LogMissingVariables()
         {
-            var missingVariables = Enum.GetValues(typeof(ExternalVariable)).Cast<ExternalVariable>()
+            var missingVariables = Enum.GetValues(typeof(ExternalVariable))
+                                       .Cast<ExternalVariable>()
                                        .Select(prop => EnvironmentVariableAttribute.Get(prop))
                                        .Where(attr => Environment.GetEnvironmentVariable(attr.Name) == null)
                                        .ToList();
 
             if (!missingVariables.Any())
                 return;
 
-            Log.Warn($"The following environment variables could not be found: " +
-                     $"\n{string.Join("\n", missingVariables.Select(var => $" - {var.Name}\t\tSource: {var.LastPassName}"))}" +
-                     $"\n\nTests that rely on these variables are likely to fail.");
+            Log.Warn($"The following environment variables could not be found: " + $"\n{string.Join("\n", missingVariables.Select(var => $" - {var.Name}\t\tSource: {var.LastPassName}"))}" + $"\n\nTests that rely on these variables are likely to fail.");
```

Comment:
Classic auto-formatter.

I'm willing to bet there is some dotsettings configuration you could change (in the shared settings) that will keep these spread across lines.

IMO it's worth taking a peek at that, or failing that manually reverting this change to the more readable original.

Reply:
rolled back - haven't touched autoformatter at this stage.